mozci/push.py

```diff
@@ -343,6 +375,47 @@ def add(result):
 
         return [Task.create(**task) for task in normalized_tasks]
 
+    def gather_missing_results(self, tasks: list) -> list:
+        """ It gathers errors/results from Taskcluster if missing."""
+        # Calling task.results modifies the properties of the task, thus, the returning list
+        # of this function comes back modified
+        future_to_task = {
+            Push.THREAD_POOL_EXECUTOR.submit(lambda task: task.results, task): task
+            for task in tasks
+            if isinstance(task, TestTask)
+        }
+
+        # XXX: Using threading even for tasks that have data in AD can be slow. This push we're testing
+        # has more than 95% of all tasks in AD. If in test_integration you comment out lines 26 and
+        # lines 165-167 you will see that it takes 4 seconds with caching (it was almost instant in the previous iteration)
+        # and 6 seconds when there's no caching (This means fetching the TC data takes 2 seconds; about 70 tasks).
+        # I'm simply calling out FYI but I think it is fine
```

Comment:
I wonder why, the threads for the tasks that already have data from AD should be super-fast... Anyway, yeah, I guess it's fine